From f48a011d9a0ca692ba7846833ffd8a362cde7558 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Sat, 7 Jun 2014 16:19:48 +1000
Subject: [PATCH 10/13] Fix leak in display subtitle code.

172 bytes would be leaked after each new subtitles in DVD, and 160 bytes with other types of subtitles.
---
 mythtv/libs/libmythtv/subtitlereader.cpp |    9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/mythtv/libs/libmythtv/subtitlereader.cpp b/mythtv/libs/libmythtv/subtitlereader.cpp
index 1fc1999..a645756 100644
--- a/mythtv/libs/libmythtv/subtitlereader.cpp
+++ b/mythtv/libs/libmythtv/subtitlereader.cpp
@@ -87,14 +87,7 @@ void SubtitleReader::ClearAVSubtitles(void)
 
 void SubtitleReader::FreeAVSubtitle(const AVSubtitle &subtitle)
 {
-    for (std::size_t i = 0; i < subtitle.num_rects; ++i)
-    {
-         AVSubtitleRect* rect = subtitle.rects[i];
-         av_free(rect->pict.data[0]);
-         av_free(rect->pict.data[1]);
-    }
-    if (subtitle.num_rects > 0)
-        av_free(subtitle.rects);
+    avsubtitle_free(const_cast<AVSubtitle*>(&subtitle));
 }
 
 void SubtitleReader::LoadExternalSubtitles(const QString &subtitleFileName,
-- 
1.7.10.2

